src/concrete/ml/sklearn/base.py

```diff
@@ -1690,3 +1694,371 @@ def predict_proba(self, X: Data, fhe: Union[FheMode, str] = FheMode.DISABLE) ->
+                    x = scatter1d(x, a + b - max_x, range_i)
+                    # x[range_i + d] = min(a, b): Second bitonic sequence gets max(a, b)
+                    x = scatter1d(x, max_x, range_i + d)
+
+                    # Max index selection
+                    sign = diff < 0
+                    max_idx = a_i + (b_i - a_i) * sign
+
+                    # Update indexes array according to the max items
+                    # idx[range_i] = a_i + b_i - max_idx <=> min_idx
+                    idx = scatter1d(idx, a_i + b_i - max_idx, range_i)
+                    # idx[range_i + d] = max_idx
+                    idx = scatter1d(idx, max_idx, range_i + d)
+
+                    # Update
+                    comparisons[range_i + d] = comparisons[range_i + d] + 1
+                    d = q - p
+                    r = p
+
+            # Return only the topk indexes
+            topk_indexes = []
+            for i in range((self.n_neighbors)):
+                topk_indexes.append(idx[i])
+
+            topk_indexes = fhe_array(topk_indexes)
+
+            return topk_indexes
+
+        # 1. Pairwise_euclidiean distance
+        # from concrete import fhe
+        # with fhe.tag(f"distance_matrix"):
+        distance_matrix = pairwise_euclidean_distance(q_X)
+
+        # The square root in the Euclidean distance calculation is not applied to speed up FHE
+        # computations.
+        # Being a monotonic function, it does not affect the logic of the calculation, notably for
+        # the argsort.
+
+        # 2. Sorting args
+        # with fhe.tag(f"sorted_args"):
+
+        sorted_args = topk_sorting(distance_matrix.flatten())
+
+        return numpy.expand_dims(sorted_args, axis=0)
+
+    # KNN works only for MONO in the latest concrete Python version
+    # FIXME: https://github.com/zama-ai/concrete-ml-internal/issues/3978
+    def compile(self, *args, **kwargs) -> Circuit:
+        # If a configuration instance is given as a positional parameter, set the strategy to
+        # multi-parameter
+        if len(args) >= 2:
+            configuration = force_mono_parameter_in_configuration(args[1])
+            args_list = list(args)
+            args_list[1] = configuration
+            args = tuple(args_list)
+
+        # Else, retrieve the configuration in kwargs if it exists, or create a new one, and set the
+        # strategy to multi-parameter
+        else:
+            configuration = kwargs.get("configuration", None)
+            kwargs["configuration"] = force_mono_parameter_in_configuration(configuration)
+
+        return BaseEstimator.compile(self, *args, **kwargs)
+
+    def predict(self, X: Data, fhe: Union[FheMode, str] = FheMode.DISABLE) -> numpy.ndarray:
+
+        X = check_array_and_assert(X)
+
+        y_preds = []
+        for query in X:
+            # Argsort
+            arg_sort = super().predict(query[None], fhe)
+            # Majority vote
+            # pylint: disable-next=protected-access
```

Comment:
are you sure you need this pylint ? we are still in the class so I don't think it should complain that you are accessing `self._y` !